kernel/src/device/tdxguest/mod.rs

```diff
@@ -168,6 +177,42 @@ struct TdxQuoteHdr {
     out_len: u32,
 }
 
+/// TDX Report structure (`TDREPORT_STRUCT`) as defined in the Intel TDX Module Specification.
+#[repr(C)]
+struct TdReport {
+    report_mac: ReportMac,
+    _reserved: [u8; 256],
+    td_info: TdInfo,
+}
```

Comment:
Add an assertion to verify the error-prone struct definition is actually correct.

```rust
const EXPECTED_SIZE_OF_TD_REPORT: usize = XXX;
const_assert!(size_of::<TdReport> == EXPECTED_SIZE_OF_TD_REPORT);
```